### PR TITLE
qa: vstart_runner initilize parent ceph_manager

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: https://tracker.ceph.com/issues/50373
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -769,23 +768,10 @@ def _set_fuse_daemon_pid(self, check_status):
...
-        # Deliberately skip parent init, only inheriting from it to get
-        # util methods like osd_dump that sit on top of raw_cluster_cmd
-        self.controller = LocalRemote()
-
-        # A minority of CephManager fns actually bother locking for when
-        # certain teuthology tests want to run tasks in parallel
-        self.lock = threading.RLock()
-
+    def __init__(self, controller, ctx):
+        super(LocalCephManager, self).__init__(controller, ctx)
```
Comment (tchaikov) [inline qa/tasks/vstart_runner.py:772]:
```suggestion
        super().__init__(controller, ctx)
```

Comment (lxbsz) [reply qa/tasks/vstart_runner.py]: Done. Thanks.

**Comment by tchaikov** [comment]
jenkins test api

File: qa/tasks/vstart_runner.py [hunk 2]
```diff
@@ -769,23 +768,10 @@ def _set_fuse_daemon_pid(self, check_status):
...
-        # Deliberately skip parent init, only inheriting from it to get
-        # util methods like osd_dump that sit on top of raw_cluster_cmd
-        self.controller = LocalRemote()
-
-        # A minority of CephManager fns actually bother locking for when
-        # certain teuthology tests want to run tasks in parallel
-        self.lock = threading.RLock()
-
+    def __init__(self, controller, ctx):
+        super().__init__(controller, ctx)
```
Comment (batrick) [inline qa/tasks/vstart_runner.py:772]: I'm not sure this won't break a lot of tests. @rishabh-d-dave do you have any thoughts?

Comment (rishabh-d-dave) [reply qa/tasks/vstart_runner.py:772]: I do intend to test this PR myself but I haven't been able to get a healthy vstart cluster. I'll get back to this as soon as I can get one.

File: qa/tasks/vstart_runner.py [hunk 3]
```diff
@@ -809,6 +795,16 @@ def run_ceph_w(self, watch_channel=None, shell=False):
...
+    def do_rados(self, cmd, pool=None, namespace=None, remote=None, **kwargs):
+        args = ['rados', '--cluster', self.cluster]
+        if pool is not None:
+            args += ['--pool', pool]
+        if namespace is not None:
+            args += ['--namespace', namespace]
+        args.extend(cmd)
+        proc = self.controller.run(args=args, wait=False, **kwargs)
+        return proc
+
```
Comment (rishabh-d-dave) [inline qa/tasks/vstart_runner.py:807]: You can do something like this and avoid duplicating `do_rados()` - https://github.com/ceph/ceph/pull/38481/files#diff-203644a2b9fd4a8cf54b0c52970ffbd1aebb87a134dcf9fc0dc88286da7d4ab6R1511.

I can do that too when I rebase my PR.

File: qa/tasks/vstart_runner.py (hunk 2 above)
Comment (lxbsz) [reply qa/tasks/vstart_runner.py:772]: Yeah, sure, will fix it.

File: qa/tasks/vstart_runner.py (hunk 3 above)
Comment (lxbsz) [reply qa/tasks/vstart_runner.py:807]: Maybe do that in your PR is better. Will leave this to you.

Comment (rishabh-d-dave) [reply qa/tasks/vstart_runner.py:812]: Okay.

File: qa/tasks/vstart_runner.py [hunk 4]
```diff
@@ -767,22 +767,15 @@ def _set_fuse_daemon_pid(self, check_status):
...
         self.lock = threading.RLock()
-
-        self.log = lambda x: log.debug(x)
-
-        # Don't bother constructing a map of pools: it should be empty
-        # at test cluster start, and in any case it would be out of date
-        # in no time.  The attribute needs to exist for some of the CephManager
-        # methods to work though.
         self.pools = {}
+        self.log = lambda x: log.debug(x)
```
Comment (rishabh-d-dave) [inline qa/tasks/vstart_runner.py:778]: [1]

**Review by rishabh-d-dave — CHANGES_REQUESTED** [review]
Looks good otherwise. Testing too went fine.

The PR title is outdated. Perhaps, you want to change it for your convenience?  :)

**Pull request closed without merging**